src/unix_dgram.cc

```diff
@@ -334,20 +334,29 @@ NAN_METHOD(Connect) {
   sockaddr_un s;
   int err;
   int fd;
+  socklen_t namelen;
+  unsigned int len;
 
   assert(info.Length() == 2);
 
   fd = info[0]->Int32Value();
   String::Utf8Value path(info[1]);
+  len = path.length();
+  if (len > sizeof(s.sun_path)) {
+    err = -EINVAL;
+    goto out;
+  }
 
   memset(&s, 0, sizeof(s));
-  strncpy(s.sun_path, *path, sizeof(s.sun_path) - 1);
+  memcpy(s.sun_path, *path, len);
```

Comment:
Maybe including the size check to the `bind` and `send` methods is a good idea too?